teps/0060-remote-resource-resolution.md

```diff
@@ -242,207 +255,570 @@ users can leverage those shared pipelines.
+for more on specific resolvers.
+
+The `ResourceRequest` Lifecycle Controller will be responsible for the
+following:
+1. Initialize `ResourceRequest` status fields on object creation.
+2. Observe `ResourceRequest` objects for populated `status.data` field
+   and set their `Succeeded` condition to `"True"`.
+3. Enforce global timeouts for all `ResourceRequests`, marking them
+   failed if the current time minus their `metadata.creation_timestamp`
+   is longer than the limit.
 
 ### Risks and Mitigations
 
-<!--
-What are the risks of this proposal and how do we mitigate. Think broadly.
-For example, consider both security and how this will impact the larger
-kubernetes ecosystem.
+#### Relying on a CRD as storage for in-lined resolved data
 
-How will security be reviewed and by whom?
+_Risk_: Relying on a CRD may introduce scaling problems that couldn't be
+discovered during proof-of-concept testing. Task and Pipeline
+definitions may only get larger until a CRD no longer provides enough
+space. In busy CI/CD clusters many rapidly created large
+`ResourceRequests` may cause API server or database performance to
+degrade.
 
-How will UX be reviewed and by whom?
+_Possible Mitigation_: Implement a `PipelineResource`-style artifact
+system where short-lived PVCs or an object store are used as ephemeral
+intermediary storage. A `ResourceRequest` could store a tuple of
+(`status.dataRef`, `status.dataDigest`) in place of in-lined
+`status.data`.  `status.dataRef` would be a pointer to the PVC or object
+store and `status.dataDigest` would be a low collision hash of the data
+being pointed to for Pipelines to verify when it "follows the pointer"
+and downloads the data.
 
-Consider including folks that also work outside the WGs or subproject.
--->
+#### Changing the way it works means potentially rewriting multiple resolvers
+
+_Risk_: Changes to the way Remote Resolution is implemented during alpha
+mean that all resolvers at some point need to be updated or rewritten as
+well.
+
+_Possible Mitigation_: To help mitigate this the alpha resolvers will be
+written to be agnostic about the "protocol". All interaction with
+clients / CRDs will be kept in shared helpers so that a rewrite only
+impacts that shared code.
 
 ### User Experience (optional)
 
-<!--
-Consideration about the user experience. Depending on the area of change,
-users may be task and pipeline editors, they may trigger task and pipeline
-runs or they may be responsible for monitoring the execution of runs,
-via CLI, dashboard or a monitoring system.
+#### Simple flow: user submits `TaskRun` using public catalog `Task`
 
-Consider including folks that also work on CLI and dashboard.
--->
+1. User applies a `TaskRun` with a `taskRef` that points to a remote:
 
-### Performance (optional)
+```yaml
+kind: TaskRun
+metadata:
+  name: my-tr
+spec:
+  taskRef:
+    resolver: git
+    resource:
+      repository_url: https://github.com/tektoncd/
```

Comment:
nit: wrong repository url 😝, it ain't a repo 👼🏼

Reply:
Well spotted, fixed cheers!